fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/runners/throttled/ThrottledIteratorTest.java

```diff
@@ -577,6 +580,65 @@ void testWithRealRecords() throws Exception {
         assertThat(itemsScanned).isEqualTo(IntStream.range(0, numRecords).boxed().collect(Collectors.toList()));
     }
 
+    @ParameterizedTest
+    @BooleanSource
+    void testRollBackTransaction(boolean commitWhenDone) throws Exception {
+        final int numRecords = 50;
+
+        final CursorFactory<Tuple> cursorFactory = (store, lastResult, rowLimit) -> {
```

Comment:
I think if you convert this to a method it could get some reuse in other tests `scanRecordKeys(store, lastResult, rowLimit)`

Reply:
Sure, done.